src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -0,0 +1,1876 @@
+    }
+
+    std::uint32_t closeTime = roundCloseTime (
+        mOurPosition->getCloseTime (), mCloseResolution);
+    bool closeTimeCorrect = true;
+
+    if (closeTime == 0)
+    {
+        // we agreed to disagree
+        closeTimeCorrect = false;
+        closeTime = mPreviousLedger->getCloseTimeNC () + 1;
+    }
+
+    WriteLog (lsDEBUG, LedgerConsensus)
+        << "Report: Prop=" << (mProposing ? "yes" : "no")
+        << " val=" << (mValidating ? "yes" : "no")
+        << " corLCL=" << (mHaveCorrectLCL ? "yes" : "no")
+        << " fail=" << (mConsensusFail ? "yes" : "no");
+    WriteLog (lsDEBUG, LedgerConsensus)
+        << "Report: Prev = " << mPrevLedgerHash
+        << ":" << mPreviousLedger->getLedgerSeq ();
+    WriteLog (lsDEBUG, LedgerConsensus)
+        << "Report: TxSt = " << set->getHash ()
+        << ", close " << closeTime << (closeTimeCorrect ? "" : "X");
+
+    // Put failed transactions into a deterministic order
+    CanonicalTXSet retriableTransactions (set->getHash ());
+
+    // Build the new last closed ledger
+    auto newLCL = std::make_shared<Ledger> (false, *mPreviousLedger);
+
+    // Set up to write SHAMap changes to our database,
+    //   perform updates, extract changes
+    WriteLog (lsDEBUG, LedgerConsensus)
+        << "Applying consensus set transactions to the"
+        << " last closed ledger";
+    applyTransactions (set, newLCL, newLCL, retriableTransactions, false);
+    newLCL->updateSkipList ();
+    newLCL->setClosed ();
+
+    int asf = newLCL->peekAccountStateMap ()->flushDirty (
+        hotACCOUNT_NODE, newLCL->getLedgerSeq());
+    int tmf = newLCL->peekTransactionMap ()->flushDirty (
+        hotTRANSACTION_NODE, newLCL->getLedgerSeq());
+    WriteLog (lsDEBUG, LedgerConsensus) << "Flushed " << asf << " accounts and " <<
+        tmf << " transaction nodes";
+
+    // Accept ledger
+    newLCL->setAccepted (closeTime, mCloseResolution, closeTimeCorrect);
+
+    // And stash the ledger in the ledger master
+    if (getApp().getLedgerMaster().storeLedger (newLCL))
+        WriteLog (lsDEBUG, LedgerConsensus)
+            << "Consensus built ledger we already had";
+    else if (getApp().getInboundLedgers().find (newLCL->getHash()))
+        WriteLog (lsDEBUG, LedgerConsensus)
+            << "Consensus built ledger we were acquiring";
+    else
+        WriteLog (lsDEBUG, LedgerConsensus)
+            << "Consensus built new ledger";
+
+    WriteLog (lsDEBUG, LedgerConsensus)
+        << "Report: NewL  = " << newLCL->getHash ()
+        << ":" << newLCL->getLedgerSeq ();
+    uint256 newLCLHash = newLCL->getHash ();
+    // Tell directly connected peers that we have a new LCL
+    statusChange (protocol::neACCEPTED_LEDGER, *newLCL);
+
+    if (mValidating && !mConsensusFail)
+    {
+        // Build validation
+        auto v = std::make_shared<STValidation> (newLCLHash,
+            consensus_.validationTimestamp (), mValPublic, mProposing);
```

Comment:
This call changed from `getApp().getOPs().getValidationTimeNC()` to `consensus_.validationTimestamp()`.  I'm fine with that, but I'm pointing out to make sure that you intended to make that change.